scripts/irods/test/test_ichmod.py

```diff
@@ -1,28 +1,74 @@
+from . import session
 from .. import lib
 
-class Test_iChmod(ResourceBase, unittest.TestCase):
-
+class Test_ichmod(session.make_sessions_mixin([('otherrods', 'rods')], [('alice', 'apass')]), unittest.TestCase):
     def setUp(self):
-        super(Test_iChmod, self).setUp()
+        super(Test_ichmod, self).setUp()
+
+        self.admin = self.admin_sessions[0]
+        self.user = self.user_sessions[0]
 
     def tearDown(self):
-        super(Test_iChmod, self).tearDown()
+        super(Test_ichmod, self).tearDown()
 
-    def test_ichmod_r(self):
+    def test_ichmod_r_with_garbage_path(self):
         self.admin.assert_icommand('imkdir -p sub_dir1\\\\%/subdir2/')
-        self.admin.assert_icommand('ichmod read ' + self.user0.username + ' -r sub_dir1\\\\%/')
+        self.admin.assert_icommand('ichmod read ' + self.user.username + ' -r sub_dir1\\\\%/')
         self.admin.assert_icommand('ichmod inherit -r sub_dir1\\\\%/')
         filepath = os.path.join(self.admin.local_session_dir, 'file')
         lib.make_file(filepath, 1)
         self.admin.assert_icommand('iput ' + filepath + ' sub_dir1\\\\%/subdir2/')
-        self.user0.assert_icommand('iget ' + self.admin.session_collection + '/sub_dir1\\\\%/subdir2/file ' + os.path.join(self.user0.local_session_dir, ''))
+        self.user.assert_icommand('iget ' + self.admin.session_collection + '/sub_dir1\\\\%/subdir2/file ' + os.path.join(self.user.local_session_dir, ''))
+
+
+class test_collection_acl_inheritance(session.make_sessions_mixin([('otherrods', 'rods')], [('alice', 'apass')]), unittest.TestCase):
+    def setUp(self):
+        super(test_collection_acl_inheritance, self).setUp()
+
+        self.admin = self.admin_sessions[0]
+        self.user = self.user_sessions[0]
+
+        self.collection = os.path.join(self.admin.session_collection, 'sub_dir1', 'subdir2')
+        self.admin.assert_icommand(['imkdir', '-p', self.collection])
+        self.admin.assert_icommand(['ichmod', 'read', self.user.username, '-r', os.path.dirname(self.collection)])
+        self.admin.assert_icommand(['ichmod', 'inherit', '-r', os.path.dirname(self.collection)])
+
+    def tearDown(self):
+        super(test_collection_acl_inheritance, self).tearDown()
+
+    def test_collection_acl_inheritance_for_iput__issue_3032(self):
+        filepath = os.path.join(self.admin.local_session_dir, 'file')
+        lib.make_file(filepath, 1)
+
+        logical_path = os.path.join(self.collection, 'file')
+
+        # Put the data object into the inheritance structure and ensure that the user given
+        # permission to read is able to get the data object.
+        self.admin.assert_icommand(['iput', filepath, logical_path])
+        self.admin.assert_icommand(['ils', '-LA', self.collection], 'STDOUT', 'file')
+        self.user.assert_icommand(['iget', logical_path, os.path.join(self.user.local_session_dir, '')])
+
+    def test_collection_acl_inheritance_for_icp__issue_3032(self):
```

Comment:
please consider `imv` into the inheritance structure as well.

Reply:
Done

Reply:
I take it back. `imv` does not work in this scenario. That's what I get for trying to go fast